subiquity/common/filesystem/manipulator.py

```diff
@@ -165,19 +167,29 @@ def reformat(self, disk, ptable=None):
             self.delete_partition(p, True)
         self.clear(disk)
 
+    def can_resize_partition(self, partition):
+        if not partition.preserve:
+            return True
+        if partition.format not in get_resize_fstypes():
+            return False
+        return True
+
     def partition_disk_handler(self, disk, spec, *, partition=None, gap=None):
         log.debug('partition_disk_handler: %s %s %s %s',
                   disk, spec, partition, gap)
 
         if partition is not None:
-            if 'size' in spec:
+            if 'size' in spec and spec['size'] != partition.size:
                 trailing, gap_size = \
                      gaps.movable_trailing_partitions_and_gap_size(partition)
                 new_size = align_up(spec['size'])
                 size_change = new_size - partition.size
                 if size_change > gap_size:
                     raise Exception("partition size too large")
+                if not self.can_resize_partition(partition):
+                    raise Exception("partition cannot support resize")
                 partition.size = new_size
+                partition.resize = True
```

Comment:
I guess it's not a problem exactly but I feel like it might be less confusing to only set this on a preserve=True partition.

Reply:
Good point, that matches the curtin assumptions but isn't a showstopper as I have set up curtin to be flexible about non-resizes masquerading as resizes (such as a partition reformat).  I will think about it for future branches.